model/test_results.go

```diff
@@ -194,11 +194,13 @@ func (t *TestResults) Append(ctx context.Context, results []TestResult) error {
 			for result := range resultChan {
 				data, err := bson.Marshal(result)
 				if err != nil {
-					catcher.Wrap(err, "problem marshalling bson")
+					catcher.Wrapf(err, "problem marshalling bson for test result '%s'", result.TestName)
+					return
```

Comment:
Do we actually want to return in this case? It seems like this would be a continue-on-error case so that it can post as many test results as it possibly can without stopping early.

Reply:
True, we were returning in the other case. But that makes sense.